plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/application/importoperation/MavenProjectImportOperation.java

```diff
@@ -53,9 +59,38 @@ public void setFilters(Collection<String> filters) {
 		MavenModelManager modelManager = mavenPlugin.getMavenModelManager();
 		Set<MavenProjectInfo> projectInfos = getMavenProjects(getProjectDirectory(), filters, modelManager, monitor);
 		ProjectImportConfiguration projectImportConfiguration = new ProjectImportConfiguration();
-		List<IMavenProjectImportResult> importResults =
-				configurationManager.importProjects(projectInfos, projectImportConfiguration, monitor);
-		return validate(toProjects(importResults));
+		if (overwriteMavenProjects(projectInfos, projectImportConfiguration, monitor)) {
+	        List<IMavenProjectImportResult> importResults =
+	                configurationManager.importProjects(projectInfos, projectImportConfiguration, monitor);
+	        return validate(toProjects(importResults));
+		} else {
+		    return Collections.EMPTY_LIST;
+		}
+	}
+
+	private boolean overwriteMavenProjects(Collection<MavenProjectInfo> projectInfos, final ProjectImportConfiguration configuration, IProgressMonitor monitor) throws CoreException {
+	    final IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+	    Set<IProject> s1 = projectInfos.stream()
+	                                      .map(project -> configuration.getProjectName(project.getModel()))
+	                                      .map(root::getProject)
+	                                      .filter(project -> project.exists())
+	                                      .collect(Collectors.toSet());
+        boolean overwrite = true;
+	    if (!s1.isEmpty()) {
+	        final String message = (s1.size() == 1) ? NLS.bind(OpenShiftCommonUIMessages.MavenProjectWarningMessage,
+                                                               s1.iterator().next().getName())
+	                                                : OpenShiftCommonUIMessages.MavenProjectsWarningMessage; 
+                        
+	        overwrite = displayOverwriteDialog(OpenShiftCommonUIMessages.OverwriteProjectsDialogTitle, message);
+            for(IProject project : s1) {
+                if (overwrite) {
+                    project.delete(false, true, monitor);
+                } else {
+                    project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
```

Comment:
Resources on disk are already changed, there was another question dialog for that, now user has been asked if he wants to overwrite the project but that did not mean resources on disk, but exactly the project resource object itself. if user declines, he remains with old project resource referencing new resources on disk, the workspace has to be made aware of it by calling refresh.